tests/unit_tests/test_routes.py

```diff
@@ -75,3 +75,43 @@ def test_get_id_nonexistent(self):
         self.assertEqual(
             response.json['error']['message'], f'Could not find {id_to_fetch} within the database.'
         )
+
+
+class TestPutEndpoint(BaseRouteTestCase):
+
+    def test_valid_kml_put(self):
+
+        response = self.app.post(
+            "/kml", data=self.kml_string, content_type="application/vnd.google-earth.kml+xml"
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.content_type, "application/json")
+
+        id_to_fetch = response.json['id']
+
+        response = self.app.put(
+            f'/kml/{id_to_fetch}',
+            data=self.new_kml_string,
+            content_type="application/vnd.google-earth.kml+xml"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.content_type, "application/json")
+
+    def test_invalid_kml_put(self):
+
+        response = self.app.post(
+            "/kml", data=self.kml_string, content_type="application/vnd.google-earth.kml+xml"
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.content_type, "application/json")
```

Comment:
Probably also make sure, that the response.json of the initial POST and of the PUT is equal.
So you could rename `response` to `response_post` everywhere in lines 84 to 90 and add something like
`self.assertEqual(response_post.json, response.json)` below line 106